utils/result.hh

```diff
@@ -59,4 +59,24 @@ concept ResultRebindableTo =
 template<typename T, ExceptionContainerResult R>
 using rebind_result = bo::result<T, typename R::error_type, exception_container_throw_policy>;
 
+struct result_with_eptr_throw_policy : bo::policy::base {
+    template<class Impl> static constexpr void wide_value_check(Impl&& self) {
+        if (!base::_has_value(self)) {
+            std::rethrow_exception(base::_error(self));
+        }
+    }
```

Comment:
The fact that boost::outcome can be configured to throw when `value()` is called is nice, but unfortunately std::expected does not work like that and only throws `std::bad_expected_access<ErrorType>` :/

This is a preexisting issue, no need to do anything here but worth keeping in mind if we eventually decide to move to the standard library-native solution.